pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -45,42 +46,47 @@ func (r *DNSPolicyReconciler) reconcileDNSRecords(ctx context.Context, dnsPolicy
 	return nil
 }
 
-func (r *DNSPolicyReconciler) reconcileGatewayDNSRecords(ctx context.Context, gateway *gatewayv1beta1.Gateway, dnsPolicy *v1alpha1.DNSPolicy) error {
+func (r *DNSPolicyReconciler) reconcileGatewayDNSRecords(ctx context.Context, gw *gatewayv1beta1.Gateway, dnsPolicy *v1alpha1.DNSPolicy) error {
 	log := crlog.FromContext(ctx)
 
-	if err := r.dnsHelper.removeDNSForDeletedListeners(ctx, gateway); err != nil {
+	trafficGateway, err := traffic.NewGateway(gw)
```

Comment:
The traffic pkg is probably a bit of an old concept now since we don't work with anything other than Gateways. It's the right idea, use a wrapper like this, but I'm wondering we should maybe just delete the traffic package (Pretty sure it[s not sued anywhere) and put it elsewhere :thinking:

Reply:
I'd still like to keep it away from any serious packages to not beg for import cycles. Refactored it a bit to get rid of old juju :thinking: